### ISSUE TITLE
low gpu usage, 

### ISSUE THREAD

**Issue by xinqiao123** [lucidrains/tab-transformer-pytorch#15]
Hi. 

I'm having a problem with running your code with my dataset. It's pretty slow. GPU runs at 50% usage in average and each epoch takes almost 900 seconds to run.

My dataset has 590540 rows, 24 categorical features, and 192 continuous features. Categories are encoded using Label encoder. Total dataset size is around 600Mb. My gpu is an integrated NVIDIA RTX 3060 with 6Gb of RAM. Optimizer is Adam.

These are the software versions:

Windows 10

Python: 3.7.11
Pytorch: 1.7.0+cu110
Numpy: 1.21.2

Let me know if you need more info from my side.

Thanks.

Xin.

**Comment by lawrence-axb** [comment]
I would recommend you to try common dataset or few dataset or dataloader to check which factors affect GPU usage
